js-core/homeworks/homework-6/src/main.js

```diff
@@ -0,0 +1,503 @@
+
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents'));
+// ehT tnemucoD tcejbO ledoM )MOD( si a gnimmargorp ecafretni rof LMTH dna LMX stnemucod
+
+/*
+ * TASK 3
+ * Добавьте в функцию reverseEachWord второй параметр,
+ * данный параметр булево, если true - тогда предложение так же
+ * переворачиваются в обратном порядке
+ * */
+
+function reverseEachWord2(str, boolean) {
+    if(boolean === true) {
+        let reverseStr = str.split(' ').reverse().join(' ');
+        return reverseEachWord(reverseStr);
+    } else {
+        return reverseEachWord(str);
+    }
+}
+
+console.log(reverseEachWord2('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM', true));
+//MOD eht fo snoitatnemelpmi tnereffid evah sresworb tnereffiD .MOD eht gnisu nigeb ot laiceps gnihtyna od ot evah t'nod uoY
+console.log(reverseEachWord2('The Document Object Model (DOM) is a programming interface for HTML and XML documents', true));
+// stnemucod LMX dna LMTH rof ecafretni gnimmargorp a si )MOD( ledoM tcejbO tnemucoD ehT
+
+console.log(reverseEachWord2('Hi my Name is', false));
+// iH ym emaN si
+
+/* TASK 3
+ * Посчитайте сколько одинаковых слов в предложении.
+ * Функция должна возвращать объект у которого свойства
+ * это повторяющиеся слово,
+ * а значение это количество повторений
+ * */
+
+// Both - Java - and - Java - Script - is - programming - and - programming - OOPBased
+
+function wordCounter(sentence) {
+    const sentenceTransformToArray = sentence.split(' ');
+    const sortSentenceArray = sentenceTransformToArray.sort();
+    let objectWithWords = {};
+    sortSentenceArray.forEach(function(word, index, arr) {
+        objectWithWords[word] = objectWithWords[word] + 1 || 1;
+    });
+    return objectWithWords;
+}
+
+console.log(
+  wordCounter('Both Java and Java Script is programming and programming OOPBased Language'),
+);
+/*
+ {
+ Both:1,
+ Java:2,
+ and:2,
+ Script: 1,
+ is: 1
+ programming: 2
+ OOPBased:1,
+ Language:1
+ }
+ */
+
+console.log(wordCounter('asd qwe asd'));
+/*
+ {
+ asd:2
+ qwe:1
+ }
+ * */
+
+console.log(wordCounter('url http url www url http'));
+
+/*
+ {
+ http:2,
+ url:3,
+ www:1
+ }
+ * */
+
+/*
+  TASK 4
+ // Функция принимает массив у которого есть свойста _id и Company.
+ // верните объект, у которого ключ это _id, а значение Company
+ */
+
+function createHashTags(arr) {
+    const objectWithIdAndCompanies = {};
+    arr.forEach(function(value, index, arr) {
+        objectWithIdAndCompanies[value._id] = value.company;
+    });
+    return objectWithIdAndCompanies;
+}
+
+console.log(createHashTags(listOfCompanys));
+//{"584babb6eeb4137cf14c37a3":"ASIMILINE", '584babb6eeb4137cf14c37a3':'Company2', }
+
+// @ SUPER
+/*
+ *
+ * TASK 1
+ * Выведите уникальные значения
+ *
+ * */
+
+function uniqueElements(arr) {
+    let newArray = [];
```

Comment:
looks very complex.

We also don't need sort here.

If you are going to create a new array, we just need to check if the same is already inside.

You have to think about **.filter** instead of using for-loop